### DISCUSSION TITLE
Squashing Github commits

### DISCUSSION THREAD

**Discussion by joelreymont** [jj-vcs/jj#5182] [General]
<img width="1339" alt="Screenshot 2024-12-24 at 4 12 04 PM" src="https://github.com/user-attachments/assets/7af72a42-e65a-468f-affc-1b144389e0cb" />

My Github commit history looks like the above.

My jj log looks like this, though 

```
@  nywuzmmy 18791+joelreymont@users.noreply.github.com 2024-12-24 16:13:08 c7af0
237
│  (no description set)
◆  xtvovsqr 18791+joelreymont@users.noreply.github.com 2024-12-23 21:20:26 maste
r 7117354f
│  wip
~
```

Is there a way for me to squash the Github commits into a single `wip` one?

**Comment by scott2000** [comment]
Right now, some of the commits are immutable, so they aren't being shown in the log, and `jj` won't let you squash them unless you pass `--ignore-immutable`. I believe you could do something like `jj squash --from cb5357a..7117354 --into cb5357a --ignore-immutable` to squash all of the GitHub "wip" commits into one.

**Comment by joelreymont** [comment]
That worked, thank you very much!
